configs/myconfigs/sample_one.py

```diff
@@ -0,0 +1,164 @@
+    by pointing to the same output directory. This script will
+    automatically create a sub-direcotry for the sample.
+
+"""
+
+import os
+import sys
+import time
+
+import m5
+from m5.objects import *
+from m5.ticks import fromSeconds
+from m5.util.convert import toLatency
+
+sys.path.append('configs/common/') # For the next line...
+import SimpleOpts
+
+from system import MySystem
+
+# Sampling library
+from sampling import forwardToSample, Sample, makeSampleDir, \
+                     checkSystem, loadSamples
+
+SimpleOpts.add_option("--script", default='',
+                      help="Script to execute in the simulated system")
+SimpleOpts.set_usage("usage: %prog [options] sampleno runs=1")
+
+
+def simulateROI(system, opts, sampleno, runs):
+    """ Fast forward to and simulate a single sample point """
+    # Create the sample directory with our friendly method
+    makeSampleDir(sampleno)
+
+    # First forward to the specific sample number
+    print "Fast forwarding to sample", sampleno
+    forwardToSample(system, sampleno)
+
+    # Take runs number of measurements at this point
+    Sample(system, opts, runs)
+    # ... and we are done!
+
+
+if __name__ == "__m5_main__":
+    (opts, args) = SimpleOpts.parse_args()
+
+    # create the system we are going to simulate
+    system = MySystem(opts)
+
+    # Check if the system is compatible with sampling library
+    checkSystem(system)
+
+    if not (len(args) == 1 or len(args) == 2):
+        SimpleOpts.print_help()
+        fatal("Simulate script requires one or two arguments")
+
+    sampleno = int(args[0])
+
+    if len(args) == 2:
+        runs = int(args[1])
+    else:
+        runs = 1
+
+    # For workitems to work correctly
+    # This will cause the simulator to exit simulation when the first work
+    # item is reached and when the first work item is finished.
+    system.work_begin_exit_count = 1
+    system.work_end_exit_count = 1
+
+    # Read in the script file passed in via an option.
+    # This file gets read and executed by the simulated system after boot.
+    # Note: The disk image needs to be configured to do this.
+    system.readfile = opts.script
+
+    # set up the root SimObject and start the simulation
+    root = Root(full_system = True, system = system)
+
+    if system.getHostParallel():
+        # Required for running kvm on multiple host cores.
+        # Uses gem5's parallel event queue feature
+        # Note: The simulator is quite picky about this number!
+        root.sim_quantum = int(1e9) # 1 ms
+
+    # Disable the gdb ports. Required for high core counts and forking.
+    m5.disableAllListeners()
+
+    # instantiate all of the objects we've created above
+    m5.instantiate()
+
+    # Read the sample points from the output directory
+    # Note: this assumes you have a file random.samples
+    # generated in the output directory of this gem5 process
+    loadSamples(m5.options.outdir + '/random.samples')
+
+
+    globalStart = time.time()
```

Comment:
You could probably break this out into a function. It's repeated in all of the run files, right?

Basically, it's always the same except for the "simulateROI" function. You could pass that in as a parameter. Let me know if you don't understand what I'm saying here and I'll give more details.